pkg/array/array.go

```diff
@@ -124,6 +128,103 @@ func (s *Locker) UpdateArrays(configPath string, fs fs.Interface) error {
 	return nil
 }
 
+type NASCooldownTracker interface {
+	MarkFailure(nas string)
+	IsInCooldown(nas string) bool
+	ResetFailure(nas string)
+	FallbackRetry(nasList []string) string
+}
+
+type NASStatus struct {
+	Failures      int
+	CooldownUntil time.Time
+}
+
+type NASCooldown struct {
+	statusMap      map[string]*NASStatus
+	cooldownPeriod time.Duration
+	threshold      int
+	mu             sync.Mutex
+}
+
+// NewNASCooldown returns a new instance of NASCooldown.
+func NewNASCooldown(cooldownPeriod time.Duration, threshold int) *NASCooldown {
+	return &NASCooldown{
+		statusMap:      make(map[string]*NASStatus),
+		cooldownPeriod: cooldownPeriod,
+		threshold:      threshold,
+		mu:             sync.Mutex{},
+	}
+}
+
+// GetStatusMap is a getter for statusMap
+func (n *NASCooldown) GetStatusMap() map[string]*NASStatus {
```

Comment:
I  think the addition of mutex locking and unlocking in getter methods (GetStatusMap, GetCooldownPeriod, GetThreshold) is required to ensure thread safety. Any thoughts?